### PR TITLE
[36243][30226] Different avatars for placeholders and groups

### PR DESCRIPTION
**Pull request by oliverguenther**
This PR:

- [x] allows rendering different avatar styles based on the type of a principal.
- [x] Extends the CreateAutocompleter as an intermediate solution as discussed to allow avatar rendering

The styles are taking as is from Wieland so I'd rather not change them in the course of this PR.

https://community.openproject.org/wp/36243
https://community.openproject.org/wp/30226/

### CODE REVIEW

**Comment by oliverguenther** [comment]
Hi @b12f , could you take a look at the frontend changes here? In the backend, the avatar helper patch you've also worked on was drastically simplified as we always render the op-principal component anyway, so we don't have to check what avatar type the user actually has.

I also ensured that the members autocompleter returns the correct hrefs we expected in the frontend so there's some minor changes there but that's otherwise only a move of where the json gets rendered in the backend.

File: frontend/src/global_styles/content/modules/_avatars.sass [hunk 1]
```diff
@@ -28,4 +28,12 @@
 
 .user-avatar--multi-line
   display: block
-  margin-bottom: 2px
\ No newline at end of file
+  margin-bottom: 2px
+
+.avatar
```
Comment (b12f) [inline frontend/src/global_styles/content/modules/_avatars.sass]: if this is a BEM block you want to keep I'd suggest renaming it to `op-avatar` and putting it into its own file

Comment (b12f) [reply frontend/src/global_styles/content/modules/_avatars.sass]: Might be hard to do right now with all the legacy stuff though

File: frontend/src/global_styles/content/modules/_avatars.sass [hunk 2]
```diff
@@ -28,4 +28,12 @@
...
+  &_placeholder_user
```
Comment (b12f) [inline frontend/src/global_styles/content/modules/_avatars.sass]:
```suggestion
  &_placeholder-user
```

An underscore denotes a modifier, of which you can only have one per class

**Review by b12f — CHANGES_REQUESTED** [review]

File: frontend/src/global_styles/content/modules/_avatars.sass (hunk 2 above)
Comment (oliverguenther) [reply frontend/src/global_styles/content/modules/_avatars.sass]: I was simply reusing the existing principal types that use underscores. Not too hard to just replace() them though. Will do

**Comment by oliverguenther** [comment]
@b12f I extracted all user avatar related styles into two BEM blocks: op-principal and op-principal-avatar and addressed your feedback. 

I removed the ` avatar-classes` input to principals as that was almost always used for setting the classes. The backend now simply uses the `size` property instead, so we can safely remove it. There were two cases that used the class to override some positioning. For them I allowed a limited override to `.op-principal-avatar` to avoid too much restyling in this PR.

I also noticed we were confusing `$href` and `href` in the value options of CreateAutocompleter and the Principal inputs. So I removed the accessor to $href  for all hal resources as it's no longer necessary.

**Review by b12f — APPROVED** [review]
LGTM

**Pull request merged**